accounts-db/src/accounts_db.rs

```diff
@@ -8019,6 +8027,10 @@ impl AccountsDb {
             assert!(reclaimed_offsets.contains_key(&expected_slot));
         }
 
+        self.clean_accounts_stats
+            .slots_cleaned
+            .fetch_add(reclaimed_offsets.len() as u64, Ordering::Relaxed);
```

Comment:
I think this is different and could guide optimizations here. We end up marking pubkeys dead in many slots. pubkey is the input to clean. we mark the pubkey dead in all older slots. The older slot may not be in the original `uncleaned_roots` list at all. We could find that we are marking accounts dead across 400k slots each time clean runs, or it may be a much smaller number. This number could guide optimization of the algorithm. I'll monitor this before I merge it. I already started it on a validator.

Reply:
I think this stat should be equal to the delta of "uncleaned_roots_len" + number of new roots with a interval.

Reply:
here we show 350k slots getting cleaned at startup.

Steady state is more interesting though, and what I was aiming for:
![image](https://github.com/user-attachments/assets/ffe7b597-bb30-4e6c-acd5-b72687d10dd5)

we have way more slots we're marking accounts dead in than we have uncleaned roots. I think uncleaned roots will coincide with flushing the write cache.

Reply:
i think part of my thinking is: are we doing something in a serial loop with 4k, 40k, or 400k entries? If so, maybe we should parallelize this.